core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1077,9 +1080,35 @@ class WorkspaceService(
 
   private def validateAclChanges(aclChanges: Set[WorkspaceACLUpdate],
                                  existingAcls: Set[WorkspaceACLUpdate],
+                                 callingUserRoles: Set[SamResourceRole],
                                  workspace: Workspace
   ): Unit = {
     val emailsBeingChanged = aclChanges.map(_.email.toLowerCase)
+    if (callingUserRoles.isEmpty) {
+      throw new InvalidWorkspaceAclUpdateException(
+        ErrorReport(StatusCodes.BadRequest, "you do not have access to change permissions for this workspace")
+      )
+    }
+    val callingUserMaxRole = callingUserRoles.flatMap(role => WorkspaceAccessLevels.withRoleName(role.value)).max
+    // Add the existingAcl entries that are being modified so we can check what we will
+    // be removing as well as what we are adding.
+    val allRolePermissionChanges =
+      aclChanges ++ existingAcls.filter(existingAcl => emailsBeingChanged.contains(existingAcl.email.toLowerCase))
+    if (callingUserMaxRole < WorkspaceAccessLevels.Owner) {
```

Comment:
This makes assumptions about the actions contained in the `workspace` roles defined in Sam's resource type configuration, which isn't ideal as the name of a role in Sam shouldn't be used for authz control. Given that, this could be modified to check actions directly to make sure the user has the necessary `share_policy::_` actions for the requested ACL changes which would be the _right_ thing. 

Still, I don't know if that's strictly necessary here, but happy to discuss. This check isn't intended to be true authorization as that is still handled by Sam -- the goal here is just to provide some guard rails and prevent ACL changes we believe are unlikely to succeed.